addOns/ascanrules/src/test/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRuleUnitTest.java

```diff
@@ -2670,6 +2670,48 @@ protected Response serve(IHTTPSession session) {
         assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
     }
 
+    @Test
+    void shouldReportXssInSriptAttackInEval() throws NullPointerException, IOException {
+        // Given
+        String test = "/shouldReportXssInSriptAttackInEval/";
+
+        this.nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected Response serve(IHTTPSession session) {
+                        String q = getFirstParamValue(session, "q");
+                        String response;
+                        if (q != null) {
+                            // Make the eye catchers fail
+                            response =
+                                    "<html><head></head><body>\n"
+                                            + "    <script>\n"
+                                            + "      eval(encodeURIComponent('"
+                                            + q
+                                            + "'));\n"
+                                            + "    </script>\n"
+                                            + "  \n"
+                                            + "\n"
+                                            + "</body></html>";
```

Comment:
```suggestion
                            response = """
                            <html><head></head><body>
                                <script>
                                   eval(encodeURIComponent('%s'));
                                </script></body></html>""".formatted(q);
```

The formatting here might not be 100%, I did it by hand. But the point is it could use a text block instead of concatenation and be formatted to include the `q` value at `%s`.

Reply:
> The formatting here might not be 100%, I did it by hand. But the point is it could use a text block instead of concatenation and be formatted to include the `q` value at `%s`.

I made the change as suggested, and also reverted the extra newlines.